install-npm.js

```diff
@@ -1,82 +1,56 @@
 #!/usr/bin/env node
-/* eslint-disable promise/prefer-await-to-callbacks */
-
-const fs = require('fs');
-const path = require('path');
-const log = require('fancy-log');
-const _ = require('lodash');
-
 
 /**
- * Because of the way npm lifecycle scripts work, on a local install, when the
- * code has not been tranpiled yet (i.e., the first time, or after the 'build'
- * directory has been deleted) the download **will** fail, and 'npm run chromedriver'
- * will need to be run.
+ * This is the `postinstall` script which:
+ * 1. Builds the project if it isn't yet built (only happens in a dev environment), and
+ * 2. Downloads Chromedriver (which can be disabled);
+ *
+ * Because `prepare` is run _after_ `postinstall`, we cannot just use `prepare` to build the project
+ * because this script depends on the project being built!
  */
 
-const BUILD_RETRIES = 200;
-const BUILD_RETRY_INTERVAL = 1000;
+const B = require('bluebird');
+
+B.config({
+  cancellation: true,
+});
+
+const {fs} = require('appium/support');
```

Comment:
I'm not sure if this peer dependency would always exist. Perhaps it would be safer to use native `fs.promises` methods here rather than imported from appium

Reply:
hmm, yeah.